airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/source.py

```diff
@@ -0,0 +1,95 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+import base64
+from typing import Any, List, Mapping, Tuple
+
+import requests
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http.auth import TokenAuthenticator
+
+from .streams import UserSettingsStream, generate_stream_classes
+
+STREAMS = generate_stream_classes()
+# from .streams import Users, Groups, Organizations, Tickets, generate_stream_classes
+
+
+class BasicApiTokenAuthenticator(TokenAuthenticator):
+    """basic Authorization header"""
+
+    def __init__(self, email: str, password: str):
+        # for API token auth we need to add the suffix '/token' in the end of email value
+        email_login = email + "/token"
+        token = base64.b64encode(f"{email_login}:{password}".encode("utf-8"))
+        super().__init__(token.decode("utf-8"), auth_method="Basic")
+
+
+class SourceZendeskSupport(AbstractSource):
```

Comment:
Could you add a docstring comment to a source with what Snapchat Marketing does. It helps understand what happened in code on a high level.

For example: """Source Intercom fetch data from messaging platform."""